### ISSUE TITLE
callback_query for users in group not working

### ISSUE THREAD

**Issue by stratus-ss** [windkh/node-red-contrib-telegrambot#382]
I have a flow that, when an alert to telegram fires, it produces the message and the inline keyboard. The user can then click the thumbs up button to silence the alert for a time. 

I have a telegram group (and later I converted it to a super group) where my user can interact with the messages as expected. However, when anyone else in the group clicks the button, no callback_query is registered. This is confusing to me because I would expect it would work for everyone or no-one. I am sure I am just missing an option somewhere in the payload, but for the life of me I cannot figure out what.

* I have tried with and without the Auto-Answer option checked in the callback node
* I've tried with and without a messageID
* I have tried with and without the `callback_query: true` set in the message payload options (and with and without the JSON.stringify)

Just to restate: Everyone can see the button. Everyone can click on the button. However, the only user that creates a response seen by the callback_query node is my own. Other users get some sort of slight animation in the button itself when it's clicked. Mobile client on IOS shows "loading"

Any help would be useful!

![image](https://github.com/windkh/node-red-contrib-telegrambot/assets/8162705/b6b461df-45ce-42ec-b84c-1af28526d4a6)

```
[{"id":"caacfdec2b7e6a16","type":"tab","label":"Alert Silences","disabled":false,"info":"","env":[]},{"id":"f7c18f49.063c4","type":"telegram sender","z":"caacfdec2b7e6a16","name":"","bot":"8cb713a482f4239b","haserroroutput":false,"outputs":1,"x":1230,"y":460,"wires":[[]]},{"id":"385615779f4b3364","type":"function","z":"caacfdec2b7e6a16","name":"FrontDoor answerCallbackQuery","func":"var text = \"Alert Silenced for 10 minutes\"\nvar options = {\n    // text : text, <-- you can set the text here or use msg.payload.content \n    show_alert : true,\n    cache_time : 10\n};\n\nmsg.payload.content = text + \" by \" + msg.payload.from.first_name;\nmsg.payload.options = options;\nmsg.payload.type = \"message\";\n\nreturn [ msg ];\n","outputs":1,"timeout":"","noerr":0,"initialize":"","finalize":"","libs":[],"x":860,"y":460,"wires":[["f7c18f49.063c4"]]},{"id":"3b460e58.b701ba","type":"switch","z":"caacfdec2b7e6a16","name":"","property":"payload.content","propertyType":"msg","rules":[{"t":"eq","v":"front_door","vt":"str"},{"t":"eq","v":"garage_double_door","vt":"str"},{"t":"eq","v":"garage_single_door","vt":"str"}],"checkall":"true","repair":false,"outputs":3,"x":310,"y":100,"wires":[["385615779f4b3364","db0d5ca2df102339"],["385615779f4b3364","a68132e2d013b75d"],["a541ba2ee8e388c4","385615779f4b3364"]]},{"id":"981d4537.1ced18","type":"telegram event","z":"caacfdec2b7e6a16","name":"","bot":"8cb713a482f4239b","event":"callback_query","autoanswer":true,"x":120,"y":100,"wires":[["3b460e58.b701ba","3fea5a95a2c70b77"]]},{"id":"3fea5a95a2c70b77","type":"debug","z":"caacfdec2b7e6a16","name":"debug 14","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"true","targetType":"full","statusVal":"","statusType":"auto","x":220,"y":220,"wires":[]},{"id":"db0d5ca2df102339","type":"api-call-service","z":"caacfdec2b7e6a16","name":"Silence Front Door Alert","server":"46f70193.13e58","version":5,"debugenabled":false,"domain":"input_boolean","service":"turn_on","areaId":[],"deviceId":[],"entityId":["input_boolean.front_door_silence"],"data":"{\"entity_id\":\"input_boolean.front_door_slience\"}","dataType":"jsonata","mergeContext":"","mustacheAltTags":false,"outputProperties":[],"queue":"none","x":810,"y":60,"wires":[["8d159284ab14f326"]]},{"id":"8d159284ab14f326","type":"delay","z":"caacfdec2b7e6a16","name":"","pauseType":"delay","timeout":"10","timeoutUnits":"minutes","rate":"1","nbRateUnits":"1","rateUnits":"second","randomFirst":"1","randomLast":"5","randomUnits":"seconds","drop":false,"allowrate":false,"outputs":1,"x":1030,"y":60,"wires":[["df60916a321172b2"]]},{"id":"df60916a321172b2","type":"api-call-service","z":"caacfdec2b7e6a16","name":"Unsilence Front Door Alert","server":"46f70193.13e58","version":5,"debugenabled":false,"domain":"input_boolean","service":"turn_off","areaId":[],"deviceId":[],"entityId":["input_boolean.front_door_silence"],"data":"{\"entity_id\":\"input_boolean.front_door_slience\"}","dataType":"jsonata","mergeContext":"","mustacheAltTags":false,"outputProperties":[],"queue":"none","x":1280,"y":60,"wires":[[]]},{"id":"a68132e2d013b75d","type":"api-call-service","z":"caacfdec2b7e6a16","name":"Silence Double Garage Door","server":"46f70193.13e58","version":5,"debugenabled":false,"domain":"input_boolean","service":"turn_on","areaId":[],"deviceId":[],"entityId":["input_boolean.garage_double_door_silence"],"data":"{\"entity_id\":\"input_boolean.garage_double_door_silence\"}","dataType":"jsonata","mergeContext":"","mustacheAltTags":false,"outputProperties":[],"queue":"none","x":800,"y":120,"wires":[["4c709645dcc6ad13"]]},{"id":"4c709645dcc6ad13","type":"delay","z":"caacfdec2b7e6a16","name":"","pauseType":"delay","timeout":"10","timeoutUnits":"minutes","rate":"1","nbRateUnits":"1","rateUnits":"second","randomFirst":"1","randomLast":"5","randomUnits":"seconds","drop":false,"allowrate":false,"outputs":1,"x":1030,"y":120,"wires":[["6f68aab9f52060e7"]]},{"id":"6f68aab9f52060e7","type":"api-call-service","z":"caacfdec2b7e6a16","name":"Unsilence  Double Garage Door","server":"46f70193.13e58","version":5,"debugenabled":false,"domain":"input_boolean","service":"turn_off","areaId":[],"deviceId":[],"entityId":["input_boolean.garage_double_door_silence"],"data":"{\"entity_id\":\"input_boolean.garage_double_door_silence\"}","dataType":"jsonata","mergeContext":"","mustacheAltTags":false,"outputProperties":[],"queue":"none","x":1290,"y":120,"wires":[[]]},{"id":"5f2dec5151dfaaba","type":"api-call-service","z":"caacfdec2b7e6a16","name":"Unsilence  Single Garage Door","server":"46f70193.13e58","version":5,"debugenabled":false,"domain":"input_boolean","service":"turn_off","areaId":[],"deviceId":[],"entityId":["input_boolean.garage_double_door_silence"],"data":"{\"entity_id\":\"input_boolean.garage_single_door_silence\"}","dataType":"jsonata","mergeContext":"","mustacheAltTags":false,"outputProperties":[],"queue":"none","x":1290,"y":180,"wires":[[]]},{"id":"a541ba2ee8e388c4","type":"api-call-service","z":"caacfdec2b7e6a16","name":"Silence Single Garage Door","server":"46f70193.13e58","version":5,"debugenabled":false,"domain":"input_boolean","service":"turn_on","areaId":[],"deviceId":[],"entityId":["input_boolean.garage_single_door_silence"],"data":"{\"entity_id\":\"input_boolean.garage_single_door_silence\"}","dataType":"jsonata","mergeContext":"","mustacheAltTags":false,"outputProperties":[],"queue":"none","x":800,"y":180,"wires":[["7f5ed5efd08efd5d"]]},{"id":"7f5ed5efd08efd5d","type":"delay","z":"caacfdec2b7e6a16","name":"","pauseType":"delay","timeout":"10","timeoutUnits":"minutes","rate":"1","nbRateUnits":"1","rateUnits":"second","randomFirst":"1","randomLast":"5","randomUnits":"seconds","drop":false,"allowrate":false,"outputs":1,"x":1030,"y":180,"wires":[["5f2dec5151dfaaba"]]},{"id":"31f4763d1d51e72f","type":"server-state-changed","z":"caacfdec2b7e6a16","name":"Front Door","server":"46f70193.13e58","version":5,"outputs":2,"exposeAsEntityConfig":"","entityId":"binary_sensor.front_door_sensor_contact_2","entityIdType":"exact","outputInitially":false,"stateType":"str","ifState":"on","ifStateType":"str","ifStateOperator":"is","outputOnlyOnStateChange":true,"for":0,"forType":"num","forUnits":"minutes","ignorePrevStateNull":false,"ignorePrevStateUnknown":false,"ignorePrevStateUnavailable":true,"ignoreCurrentStateUnknown":true,"ignoreCurrentStateUnavailable":true,"outputProperties":[{"property":"payload","propertyType":"msg","value":"","valueType":"entityState"},{"property":"data","propertyType":"msg","value":"","valueType":"eventData"},{"property":"topic","propertyType":"msg","value":"","valueType":"triggerId"}],"x":80,"y":620,"wires":[["6976a7db0afbe64a","eafb85656b267b7a"],["6976a7db0afbe64a","eafb85656b267b7a"]]},{"id":"6976a7db0afbe64a","type":"trigger","z":"caacfdec2b7e6a16","name":"","op1":"","op2":"0","op1type":"nul","op2type":"str","duration":"2","extend":false,"overrideDelay":false,"units":"min","reset":"off","bytopic":"all","topic":"topic","outputs":1,"x":380,"y":660,"wires":[["5f83c0545f189748"]]},{"id":"a869b16cc6806d50","type":"telegram sender","z":"caacfdec2b7e6a16","name":"","bot":"8cb713a482f4239b","haserroroutput":false,"outputs":1,"x":1430,"y":640,"wires":[["8481b731d73fe0fb"]]},{"id":"eafb85656b267b7a","type":"debug","z":"caacfdec2b7e6a16","name":"","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"true","targetType":"full","statusVal":"","statusType":"auto","x":300,"y":760,"wires":[]},{"id":"5f83c0545f189748","type":"api-current-state","z":"caacfdec2b7e6a16","name":"Door Still Open?","server":"46f70193.13e58","version":3,"outputs":2,"halt_if":"on","halt_if_type":"str","halt_if_compare":"is","entity_id":"binary_sensor.front_door_sensor_contact_2","state_type":"str","blockInputOverrides":false,"outputProperties":[{"property":"payload","propertyType":"msg","value":"","valueType":"entityState"},{"property":"data","propertyType":"msg","value":"","valueType":"entity"}],"for":"0","forType":"num","forUnits":"minutes","override_topic":false,"state_location":"payload","override_payload":"msg","entity_location":"data","override_data":"msg","x":580,"y":640,"wires":[["44fb0d54babc1640"],[]]},{"id":"48894fa8cf920c2b","type":"function","z":"caacfdec2b7e6a16","name":"","func":"var opts = {\n  reply_markup: JSON.stringify({\n    \"inline_keyboard\": [[\n                {\n                    \"text\": \"\\uD83D\\uDC4D\",\n                    \"callback_data\": \"front_door\",\n                    \"selective\": false            \n                }, \n                ]\n            ]\n  }),\n  options: JSON.stringify({\n    \"callback_query\": true\n  })\n};\n\nmsg.payload.content = \"Front Door Open For More Than 2 Minutes\\nSilence Alert 10 minutes?\";\nmsg.payload.options = opts;\nmsg.payload.type = \"message\";\nmsg.payload.chatId = \"<CHATID_REMOVED>\";\nreturn [ msg ];","outputs":1,"timeout":"","noerr":0,"initialize":"","finalize":"","libs":[],"x":1240,"y":640,"wires":[["a869b16cc6806d50"]]},{"id":"1c91537d72c0cda9","type":"change","z":"caacfdec2b7e6a16","name":"","rules":[{"t":"set","p":"payload","pt":"msg","to":"{\"type\":\"message\",\"content\":\"test\",\"chatId\":\"<CHATID_REMOVED>\",\"text\":\"{{msg.payload}}\"}","tot":"json"}],"action":"","property":"","from":"","to":"","reg":false,"x":1000,"y":640,"wires":[["48894fa8cf920c2b"]]},{"id":"44fb0d54babc1640","type":"api-current-state","z":"caacfdec2b7e6a16","name":"Alert Silenced?","server":"46f70193.13e58","version":3,"outputs":2,"halt_if":"on","halt_if_type":"str","halt_if_compare":"is","entity_id":"input_boolean.front_door_silence","state_type":"str","blockInputOverrides":false,"outputProperties":[{"property":"payload","propertyType":"msg","value":"","valueType":"entityState"},{"property":"data","propertyType":"msg","value":"","valueType":"entity"}],"for":"0","forType":"num","forUnits":"minutes","override_topic":false,"state_location":"payload","override_payload":"msg","entity_location":"data","override_data":"msg","x":780,"y":640,"wires":[[],["1c91537d72c0cda9"]]},{"id":"be3ba44e40aaa0dd","type":"inject","z":"caacfdec2b7e6a16","name":"","props":[{"p":"payload"},{"p":"topic","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"","payloadType":"date","x":540,"y":820,"wires":[["44fb0d54babc1640"]]},{"id":"8481b731d73fe0fb","type":"debug","z":"caacfdec2b7e6a16","name":"","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"true","targetType":"full","statusVal":"","statusType":"auto","x":1630,"y":660,"wires":[]},{"id":"8cb713a482f4239b","type":"telegram bot","botname":"vailhomebot","usernames":"","chatids":"106842723","baseapiurl":"","updatemode":"polling","pollinterval":"300","usesocks":false,"sockshost":"","socksport":"6667","socksusername":"anonymous","sockspassword":"","bothost":"","botpath":"","localbotport":"8443","publicbotport":"8443","privatekey":"","certificate":"","useselfsignedcertificate":false,"sslterminated":false,"verboselogging":false},{"id":"46f70193.13e58","type":"server","name":"Home Assistant","version":5,"addon":true,"rejectUnauthorizedCerts":true,"ha_boolean":"y|yes|true|on|home|open","connectionDelay":true,"cacheJson":true,"heartbeat":false,"heartbeatInterval":30,"areaSelector":"friendlyName","deviceSelector":"friendlyName","entitySelector":"friendlyName","statusSeparator":"at: ","statusYear":"hidden","statusMonth":"short","statusDay":"numeric","statusHourCycle":"h23","statusTimeFormat":"h:m","enableGlobalContextStore":true}]
```

**Comment by windkh** [comment]
@stratus-ss 
Hi Steve, 
well I just tried to reproduce your scenario: created a super group, added the bot and two persons (me and my wife).
I triggered a inline keyboard and the callback_query is called if I or my wife click on a button.

I just used the standard example next to your code example: both work

![image](https://github.com/windkh/node-red-contrib-telegrambot/assets/10910767/5f81fc6d-8103-4d31-8d73-66a43fbf4896)

In my supergroup me and the bot are admin, my wife not.
I can't reproduce your problem... maybe we need to have a look at your flow together in a teams meeting?

**Comment by stratus-ss** [comment]
That would be fantastic (although I don't expect such service for something you are giving away!) How should we proceed?

**Comment by windkh** [comment]
Maybe in the evening.

**Comment by stratus-ss** [comment]
Just as an update to this

I have now tried:

* creating a new group and readding people
* having my wife create a group and adding people
* creating a new bot in telegram
* creating a new bot in node red

After you had successfully tested my situation, I thought perhaps it was a group config or old bot or something strange like that but that doesn't appear to be the case

**Comment by windkh** [comment]
Hm strange... I only used a group and accounts that I created long time ago. So does that mean the problem is solved for you?

**Comment by stratus-ss** [comment]
no the problem persists. My wife clicks the button on her phone it just says "Loading" if she clicks it on the desktop the button flashes but does nothing

**Comment by windkh** [comment]
I will be svailable in about 4 hours ... we can make a meeting there if that is notvtoo late for you

**Comment by stratus-ss** [comment]
Thank you for the offer. In 4 hours I will be on a plane. I am in GMT -5 time zone. So assuming you are in GMT +2, it will probably have to wait until the weekend or next week. 

In there is anything I can provide (screen recordings or similar) I am happy to provide those so we can talk asynchronously

**Comment by windkh** [comment]
Ok then let's try on sunday

**Comment by stratus-ss** [comment]
I can be available most times during the week. Whenever is convenient for you. Es ist nicht wichtig. Es ist nur ein kleines dinge. Und ich bin dankbar dass Sie mit mir sprechen werden.

**Comment by windkh** [comment]
@stratus-ss sorry for delay... I did not have any free slot at the weekend. Maybe tomorrow.

**Comment by windkh** [comment]
@stratus-ss can you write me an email so that I can send you a teams link?

**Comment by stratus-ss** [comment]
sure no problem

**Comment by stratus-ss** [comment]
did you receive an email @windkh ? I sent the email to the email address attached to your public github profile

**Comment by stratus-ss** [comment]
Hi @windkh I think my emails must be getting lost under the pile of emails you must receive

**Comment by windkh** [comment]
Hi Steve,
sorry, but I was on a business trip the whole week... I did not find any time to take care of you. Let's try next week...

**Comment by stratus-ss** [comment]
please feel no obligation!

I only mentioned here because I assume email gets lost like last time

**Comment by stratus-ss** [comment]
I wanted to follow up on this. I'll start by listing what I tried (and thanks to @windkh for being gracious with his time!)

**Created a new bot**

I tried with the existing channel but I created a new bot. I ended up having the similar problem. Button shows up but no callback query is registered.
Tried to remove the "Auto reply" option from the callback_query node. Checked or unchecked seemed to make no difference

**Created a new group**

Same as above, neither bot received call back query from any other user than my own.

**Created Super group**

Same as above

** Changed Settings on both Bots via Botfather**

I will enumerate which settings where changed. None of the settings individually seemed to make a difference but I will list them in case some combination of options + the final thing I tried were required to make this work

1. Turned on Inline mode
2. Turned Inline Feedback to 100%
3. Turned on Allow Groups
4. Turned off Group Privacy
5. Under Group Admin Rights turned on Manage Chats

Again, toggling these individually and testing didn't yield results

** Set all users in the group to Admin **

This had no effect

**Edited the Bot in Telegram Node**

I went to investigate the bot settings and I saw there was an optional "ChatIDs" section. So I put the new chat ID in there and it worked. I then went back and put in the ID of my old chat, and that works as well

![image](https://github.com/user-attachments/assets/15f00204-2e14-420f-92fb-40fa104d6238)


So the end result is that I was able to get this working despite limited ability to debug NodeRed itself.

Thanks to @windkh  I wouldn't have gotten here without his help!
